### PR TITLE
Add dry-run of lxc-to-lxd to verify-source command

### PR DESCRIPTION
**Pull request by axw**
We embed the [lxc-to-lxd script](https://github.com/lxc/lxd/blob/master/scripts/lxc-to-lxd) in the commands package, and run it via SSH using --dry-run on the container hosts. There are some miscellaneous fixes and improvements to SSH command execution.

### CODE REVIEW

File: commands/exec.go [hunk 1]
```diff
@@ -21,38 +23,72 @@ type FlatMachine struct {
...
+
+func withStderr(w io.Writer) execOption {
+	return func(opts *execOptions) {
+		opts.stderr = w
+	}
 }
 
 // runViaSSH runs script in the remote machine with address addr.
-func runViaSSH(addr string, script, identity string) (RunResult, error) {
+func runViaSSH(addr, script string, opts ...execOption) (int, error) {
```
Comment (babbageclunk) [inline commands/exec.go:60]: Nice.

File: commands/lxc2lxd.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import (
+	"bytes"
+	"os"
+	"strings"
+
+	"github.com/juju/description"
+	"github.com/juju/errors"
+)
+
+//go:generate go run ../juju2/generate/filetoconst/filetoconst.go LXCMigrationScript ../../../lxc/lxd/scripts/lxc-to-lxd lxc2lxd_script.go 2017 commands
```
Comment (babbageclunk) [inline commands/lxc2lxd.go:15]: Aha, I'm going to steal this for my embedded Python upgrade script.

Comment (babbageclunk) [reply commands/lxc2lxd.go:15]: Oh weird, the script was already in the codebase?

**Review by babbageclunk — APPROVED** [review]
Looks awesome.

File: commands/exec.go [hunk 3]
```diff
@@ -21,38 +23,72 @@ type FlatMachine struct {
...
-	if identity != "" {
-		sshOptions.SetIdentities(identity) //
+
+	var options execOptions
+	options.stdout = os.Stdout
+	options.stderr = os.Stderr
+	// Strict host key checking must be disabled because Juju 1.25 did not
+	// populate SSH host keys.
+	options.SetStrictHostKeyChecking(ssh.StrictHostChecksNo)
+	options.SetKnownHostsFile(os.DevNull)
```
Comment (howbazaar) [inline commands/exec.go:70]: Nice.

File: commands/lxc2lxd.go (hunk 2 above)
Comment (howbazaar) [reply commands/lxc2lxd.go:15]: No, it is grabbing it from the lxd github repo that we are using for lxd support.

**Review by howbazaar — APPROVED** [review]

File: commands/lxc2lxd.go (hunk 2 above)
Comment (axw) [reply commands/lxc2lxd.go:15]: > Oh weird, the script was already in the codebase?

The filetoconst code generation helper is there already, which you can reuse. I just ran it and checked in the generated code (lxc2lxd_script.go). As Tim says, it's pointing at the script from the lxc/lxd repo.

**Pull request merged**